File: src/towd.business/CharacterTypes.vb
Imports System.Runtime.CompilerServices
Imports towd.data

Friend Module CharacterTypes
    Friend ReadOnly Descriptors As IReadOnlyDictionary(Of CharacterType, ICharacterType) =
        New List(Of ICharacterType) From
        {
            New CharacterTypeDescriptor(CharacterType.N00b, "N00b")
        }.
        ToDictionary(Function(x) x.CharacterType, Function(x) x)
    <Extension>
    Friend Function ToDescriptor(characterType As CharacterType) As ICharacterType
        Return Descriptors(characterType)
    End Function
End Module

File: web/scripts/World/enums/character_types.js
export const CharacterType = {
    N00B: "N00B"
};
class CharacterTypeDescriptor{
    static create(){
        return new CharacterTypeDescriptor();
    }
    setInitialize(initialize){
        this.initialize = initialize;
        return this;
    }
}
export const CharacterTypes= {
    [CharacterType.N00B]:
        CharacterTypeDescriptor.
            create().
            setInitialize(()=>{}),
};